graph.py

```diff
@@ -0,0 +1,57 @@
+import random
+
+
+class Graph:
+    def __init__(self):
+        """
+        graph variable is a dictionary of lists - graph is implemented as adjacency lists
+        """
+        self.graph = dict()
+        self.nodes = set()
+
+    def _topological_sort(self, root, visited, result):
+        visited.add(root)
+
+        for neighbour in self.graph[root]:
+            if neighbour not in visited:
+                result, visited = self._topological_sort(neighbour, visited, result)
+
+        result.append(root)
+
+        return result, visited
+
+    def _get_random_node(self, nodes):
```

Comment:
I believe that DFS traversal gives us a correct solution, though I need to check it.